calculator/src/main/java/com/market/banica/calculator/service/CalculatorServiceImpl.java

```diff
@@ -1,56 +1,823 @@
+                                                                                    Map<String, List<List<ProductPriceComponentsSet>>> result,
+                                                                                    ProductDto tempProduct) {
+        List<List<ProductPriceComponentsSet>> ingredientsPriceVariantsLists = new ArrayList<>();
+        for (String ingredientName : tempProduct.getIngredients().keySet()) {
+
+            ProductDto ingredient = productDtoNamesMap.get(ingredientName);
+
+            if (ingredient.getTotalPrice().equals(BigDecimal.valueOf(-1))) {
+
+                List<ProductPriceComponentsSet> compositeIngredientPriceVariants =
+                        createPriceVariantsForCompositeProduct(productDtoMap, productSpecificationMap,
+                                productPriceComponentsSetByProductIdMap, result,
+                                tempProduct, ingredient);
+
+                ingredientsPriceVariantsLists.add(compositeIngredientPriceVariants);
+
+                continue;
+            }
+
+            Set<ProductPriceComponentsSet> ingredientPriceVariantsSet =
+                    calculatePossiblePricesForProductFromProductSpecifications(productDtoMap,
+                            productSpecificationMap, tempProduct, ingredient,
+                            productPriceComponentsSetByProductIdMap);
+
+            ingredientsPriceVariantsLists.add(new ArrayList<>(ingredientPriceVariantsSet));
+        }
+        return ingredientsPriceVariantsLists;
+    }
+
+    private List<ProductPriceComponentsSet> createPriceVariantsForOrderedProduct(Map<ProductDto, Map<String, Long>> productDtoMap,
+                                                                                 Map<String, List<ProductSpecification>> productSpecificationMap,
+                                                                                 Map<Integer, ProductPriceComponentsSet> productPriceComponentsSetByProductIdMap,
+                                                                                 Map<String, List<List<ProductPriceComponentsSet>>> result,
+                                                                                 ProductDto orderedProduct, long quantity) {
+
+        Set<ProductPriceComponentsSet> simpleProductPriceVariantsSet = calculatePossiblePricesForProductFromProductSpecifications(
+                productDtoMap, productSpecificationMap, orderedProduct,
+                orderedProduct, productPriceComponentsSetByProductIdMap);
+
+        List<List<ProductPriceComponentsSet>> compositeProductPriceVariantSets = result.get(orderedProduct.getItemName());
+
+        List<ProductPriceComponentsSet> compositeIngredientPriceVariants = compositeProductPriceVariantSets
+                .stream()
+                .flatMap(Collection::stream)
+                .filter(l -> l.getProductName().equals(orderedProduct.getItemName()))
```

Comment:
" l " could be changed with something more descriptive.

Reply:
With respect to the naming convention of the team Oracle Code  Convention says:
"Variable names should be short yet meaningful. The choice of a variable name should be mnemonic- that is, designed to indicate to the casual observer the intent of its use. **One-character variable names should be avoided except for temporary "throwaway" variables.** Common names for temporary variables are i, j, k, m, and n for integers; c, d, and e for characters."

Reply:
in this case, the variable is an object, please change it to a more meaningful name

Reply:
Fixed